safep/AFEP_parse.py

```
@@ -417,6 +421,11 @@ def get_summary_statistics(args, fepruns):
         cumulative = feprun.cumulative
         dG = np.round(cumulative.BAR.f.iloc[-1] * args.RT_kcal_per_mol, 1)
         error = np.round(cumulative.BAR.errors.iloc[-1] * args.RT_kcal_per_mol, 1)
+        if (dG == 0 or dG == np.nan) and error == 0:
+            dG = np.round(cumulative.EXP.ff.iloc[-1] * args.RT_kcal_per_mol, 1)
+            error = np.nan
+            warnings.warn("BAR estimation appears to have failed. Reporting exponential estimates."
```

Ruff (0.16.0)

[warning] No explicit `stacklevel` keyword argument found

Set `stacklevel=2`

(B028)